client/src/modules/user/UserProfile/UserProfile.jsx

```diff
@@ -197,6 +199,7 @@ class UserProfile extends PureComponent {
           {this.renderPersonalInfo()}
           {this.renderContactInfo()}
           {this.renderAccountInfo()}
+          <EmailNotifications user={currentUser} />
```

Comment:
`EmailNotifications` is a class component, is connected with redux etc. Because it is not a simple "render" component you can get user data inside this component and not pass it from the parent

Reply:
yes I know, but i will have to repeat all fetch logic inside EmailNotification component 
So all code below will be repeated two times. So I think passing user data is a better fit for this situation.


```
import { getCurrentUser } from 'modules/user/model/selectors';
import { UserPropType } from 'common/constants/propTypes';

handleFetchUserDetails = () => {
    const { fetchUserDetails } = this.props;

    this.setState({ pending: true });
    this.pendingPromise = makeAbortablePromise(fetchUserDetails());
    this.pendingPromise.promise
      .then(() => this.setState({ pending: false }))
      .catch(err => {
        if (!(err instanceof AbortPromiseException)) {
          this.setState({ pending: false });
        }
      });
  };



UserProfile.propTypes = {
  currentUser: UserPropType.isRequired,

  fetchUserDetails: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  currentUser: getCurrentUser(state)
});

export default connect(
  mapStateToProps,
  { fetchUserDetails }
)(UserProfile);